cpp/tests/experimental/spatial/polygon_bounding_boxes_test.cu

```diff
@@ -0,0 +1,133 @@
+ * limitations under the License.
+ */
+
+#include <cuspatial_test/vector_equality.hpp>
+#include <cuspatial_test/vector_factories.cuh>
+
+#include <cuspatial/error.hpp>
+#include <cuspatial/experimental/geometry/box.hpp>
+#include <cuspatial/experimental/polygon_bounding_boxes.cuh>
+#include <cuspatial/vec_2d.hpp>
+
+#include <gtest/gtest.h>
+
+template <typename T>
+struct PolygonBoundingBoxTest : public ::testing::Test {
+};
+
+using cuspatial::vec_2d;
+using cuspatial::test::make_device_vector;
+
+using TestTypes = ::testing::Types<float, double>;
+
+TYPED_TEST_CASE(PolygonBoundingBoxTest, TestTypes);
+
+TYPED_TEST(PolygonBoundingBoxTest, test_empty)
+{
+  using T = TypeParam;
+
+  auto poly_offsets = make_device_vector<int32_t>({});
+  auto ring_offsets = make_device_vector<int32_t>({});
+  auto vertices     = make_device_vector<vec_2d<T>>({});
+
+  auto bboxes = rmm::device_vector<cuspatial::box<T>>(poly_offsets.size());
+
+  auto bboxes_end = cuspatial::polygon_bounding_boxes(poly_offsets.begin(),
+                                                      poly_offsets.end(),
+                                                      ring_offsets.begin(),
+                                                      ring_offsets.end(),
+                                                      vertices.begin(),
+                                                      vertices.end(),
+                                                      bboxes.begin());
+
+  EXPECT_EQ(std::distance(bboxes.begin(), bboxes_end), 0);
+}
+
+TYPED_TEST(PolygonBoundingBoxTest, test_one)
+{
+  using T = TypeParam;
+
+  auto poly_offsets = make_device_vector<int32_t>({0});
+  auto ring_offsets = make_device_vector<int32_t>({0});
+  auto vertices     = make_device_vector<vec_2d<T>>(
+    {{2.488450, 5.856625}, {1.333584, 5.008840}, {3.460720, 4.586599}, {2.488450, 5.856625}});
+
+  auto bboxes = rmm::device_vector<cuspatial::box<T>>(poly_offsets.size());
+
+  auto bboxes_end = cuspatial::polygon_bounding_boxes(poly_offsets.begin(),
+                                                      poly_offsets.end(),
+                                                      ring_offsets.begin(),
+                                                      ring_offsets.end(),
+                                                      vertices.begin(),
+                                                      vertices.end(),
+                                                      bboxes.begin());
+
+  EXPECT_EQ(std::distance(bboxes.begin(), bboxes_end), 1);
+
+  auto bboxes_expected =
+    make_device_vector<cuspatial::box<T>>({{{1.333584, 4.586599}, {3.460720, 5.856625}}});
+
+  CUSPATIAL_EXPECT_VECTORS_EQUIVALENT(bboxes, bboxes_expected);
+}
+
+TYPED_TEST(PolygonBoundingBoxTest, test_small)
+{
+  using T = TypeParam;
+
+  auto poly_offsets = make_device_vector<int32_t>({0, 1, 2, 3});
+  auto ring_offsets = make_device_vector<int32_t>({0, 4, 9, 13});
```

Comment:
It strikes me that these offsets vectors should have `n+1` entries in them, `poly_offsets` should be `[0, 1, 2, 3, 4]` and `ring_offsets` should be `[0, 4, 9, 13, 17]` in order to comply with GeoArrow offsets standard.

Reply:
Our C++ implementation doesn't require the unnecessary final offset like Arrow does.

Reply:
OK, I corrected the offsets as you suggest and updated the implementation to ignore the final offset (since it is unneeded).  I can't actually think of a case where this final offset is really needed.